homeassistant/components/sensor/wunderground.py

```diff
@@ -657,21 +654,63 @@ def __init__(self, rest, condition):
         """Initialize the sensor."""
         self.rest = rest
         self._condition = condition
+        self._state = None
+        self._attributes = {
+            ATTR_ATTRIBUTION: CONF_ATTRIBUTION,
+        }
+        self._icon = None
+        self._entity_picture = None
+        self._unit_of_measurement = self._cfg_expand("unit_of_measurement")
         self.rest.request_feature(SENSOR_TYPES[condition].feature)
+        self._long_text = None
 
     def _cfg_expand(self, what, default=None):
+        """Parse and return sensor data."""
         cfg = SENSOR_TYPES[self._condition]
         val = getattr(cfg, what)
-        try:
-            val = val(self.rest)
-        except (KeyError, IndexError) as err:
-            _LOGGER.warning("Failed to parse response from WU API: %s", err)
-            val = default
-        except TypeError:
-            pass  # val was not callable - keep original value
+        if callable(val):
+            try:
+                val = val(self.rest)
+            except (KeyError, IndexError, TypeError, ValueError) as err:
+                _LOGGER.warning("Failed to expand cfg from WU API."
+                                " Condition: %s Attr: %s Error: %s",
+                                self._condition, what, repr(err))
+                val = default
 
         return val
 
+    def _update_attrs(self):
+        """Parse and update device state attributes."""
+        attrs = self._cfg_expand("device_state_attributes", {})
+
+        self._attributes[ATTR_FRIENDLY_NAME] = self._cfg_expand(
+            "friendly_name")
+
+        if self._long_text:
+            self._attributes['long_text'] = self._long_text
+
+        for (attr, callback) in attrs.items():
+            if callable(callback):
+                try:
+                    self._attributes[attr] = callback(self.rest)
+                except (KeyError, IndexError, TypeError, ValueError) as err:
+                    _LOGGER.warning("Failed to update attrs from WU API."
+                                    " Condition: %s Attr: %s Error: %s",
+                                    self._condition, attr, repr(err))
+            else:
+                self._attributes[attr] = callback
+
+    def _update_state(self):
+        """Parse and update state."""
+        state = self._cfg_expand("value", STATE_UNKNOWN)
+
+        if isinstance(state, str) and len(state) > 255:
```

Comment:
We shouldn't change which attributes this state is stored on based on the length. Instead define which sensor types will use the longer attribute instead of the state, and create a different state value for them to use.

Which sensor types are longer than 255?

Reply:
OK. Longer text is mostly in forecast WU features, but cannot be 100% sure what we get, because WU API is not strictly defined. Alerts text are already in attributes.
So it is a question, if check for length at all (current behavior) - long state will be refused at core level. Or let in state, and truncate to 255 ?

Reply:
Unless there's a reason we expect to get long states (i.e. alerts), we should just pass it directly to core. If we start getting long results from WU, we want that to be an exception so we know we need to re-evaluate how we're handling that attribute, instead of silently handling it differently.

Reply:
OK, reverting to old behavior